modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/provider/APIApplicationProviderImpl.java

```diff
@@ -241,21 +243,44 @@ public PropertyType getPropertyType() {
 						return PropertyType.parse(listEntry.getKey());
 					}
 
+					@Override
+					public String getRelatedPropertyERC() {
+						return (String)properties.get(
+							"apiPropertyToAPIPropertiesERC");
+					}
+
 					@Override
 					public String getSourceFieldName() {
+						if (objectField == null) {
+							return "";
+						}
+
 						return objectField.getName();
 					}
 
 					@Override
 					public Type getType() {
-						Type type = _propertyTypes.get(
-							objectField.getBusinessType());
+						Type type = null;
+
+						if (objectField == null) {
```

Comment:
If you consider the objectField == null as the condition to check this code:

As a developer, I can send a request with a random `objectFieldERC` field with an `apiPropertyTpe` `singleContainer`, and **I would pass the previous validations you wrote**, as you are validating that the `objectField` does not exist (remember the objectFieldERC is random).

is that the expected behaviour?

Reply:
If you see in the APIPropertyModelListener, I've wrote several validations where a `singleContainer` can't have `objectFieldERC` and `objectRelationshipNames` values. That's the reason I am considering null `objectField`. I also tested the flow of the code and first of all, it executes the validations and later it will call the methods of APIApplicationProviderImpl